### PR TITLE
Docs: Use TableOfContents component in ToC 

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

Replaced custom table of contents component with the new `TableOfContents` component in docs site `ToC`.

BEFORE:
ezgif com-video-to-gif (1)

AFTER:
ezgif com-video-to-gif

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-6267)

### Checklist

- [n/a] Added unit and Flow Tests
- [n/a] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/Toc.js [hunk 1]
```diff
@@ -69,6 +69,13 @@ function useThrottledOnScroll(callback: null | (() => void), delay: number) {
   }, [throttledCallback]);
 }
 
+type ToCItem = {|
+  id: string,
+  label: string,
+  // eslint-disable-next-line flowtype/no-mutable-array
```
Comment (AlbertCarreras) [inline docs/docs-components/Toc.js:75]: Can we fix this?

 $ReadOnlyArray<....>

File: docs/docs-components/Toc.js [hunk 2]
```diff
@@ -122,7 +129,11 @@ export default function Toc({ cards }: Props): Node {
 
   const handleClick = (
```
Comment (AlbertCarreras) [inline docs/docs-components/Toc.js:130]: can we use named arguments? `({ hash,  event })`

File: docs/docs-components/Toc.js (hunk 1 above)
Comment (diyorbek) [reply docs/docs-components/Toc.js:75]: If we use readonly array then the algorithm [here](https://github.com/pinterest/gestalt/pull/3133/files#diff-8c97d01b6c6c7a51b95de107d690111f4e5485389b5166c8a2ff19135fc7fe4cR179) will not make sense. I tried making it readonly and I don't think coming up with "less straightforward" approach just to satisfy Flow is a good idea.

File: docs/docs-components/Toc.js (hunk 2 above)
Comment (diyorbek) [reply docs/docs-components/Toc.js:130]: yes

**Review by AlbertCarreras — APPROVED** [review]

File: docs/docs-components/Toc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/docs-components/Toc.js:75]: give it a couple thoughts, we can merge and revisit this later on as a clean up

**Pull request merged**